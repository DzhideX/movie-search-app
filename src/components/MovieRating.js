import imdb from'../images/BG_rectangle._CB1509060989_SY230_SX307_AL_.jfif';
import rotten from'../images/691px-Rotten_Tomatoes_logo.svg.png';
import meta from '../images/1280px-Metacritic_logo.svg.png';
import React from 'react';

const MovieRating = ({rating}) => {

    const renderImage = () => {
        if(rating.Source === 'Internet Movie Database'){
            return { src: imdb, alt: 'imdb' };
        }else if(rating.Source === 'Rotten Tomatoes'){
            return { src: rotten, alt: 'rotten tomatoes' };
        }else if(rating.Source === 'Metacritic'){
            return { src: meta, alt: 'metacritic' };
        }
    }

    return (
        <div data-test='movie-rating-card' className='movie-thumbnail-more-info-modal-bottom-cards__card'>
            <img data-test='movie-rating-card-image' className='movie-thumbnail-more-info-modal-bottom-cards__rating-image' src={renderImage().src} alt={renderImage().alt}/>
            <p data-test='movie-rating-card-value' className='movie-thumbnail-more-info-modal-bottom-cards__rating-value'> {rating.Value} </p>
        </div>
    );
};

export default MovieRating;