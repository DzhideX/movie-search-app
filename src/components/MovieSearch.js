import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import noImage from '../images/340719-200.png';
import Modal from 'react-modal';
import axios from 'axios';

const Movie = ({index,title,image,year,type,id}) => {

    const [movieData, setMovieData] = useState([]);
    useEffect(() => {
        axios.get(`http://www.omdbapi.com/?apikey=10d92ac&i=${id}`).then(response=> {
            setMovieData(response.data);
        });
    });
    const [hovered, setHovered] = useState(false);
    const toggleHover = () => setHovered(!hovered);
    const [modal, setModal] = useState(false);
    const openModal = () => {
        setModal(true);
    }
    const closeModal=  () => {
        setModal(false);
        setHovered(false);
    }

    return (
        <div className='movie-thumbnail'>
            {!hovered && <div className='movie-thumbnail__no-info' onMouseEnter={toggleHover}>
            {image === 'N/A' ? <img className='movie-thumbnail-no-image-icon' src={noImage} alt='not loaded'/> : <img className='movie-thumbnail__image'  src={image} alt={index}/>} 
            <p className='movie-thumbnail__title'> {title} </p>
            </div>}
            {hovered && 
            <div className='movie-thumbnail-more-info' onMouseLeave={toggleHover}> 
                <p className='movie-thumbnail-more-info__description'> Title: </p>
                <p> {title} </p>
                <p className='movie-thumbnail-more-info__description'> Year: </p>
                <p> {year} </p>
                <p className='movie-thumbnail-more-info__description'> Type: </p>
                <p> {type} </p>
                <button onClick={openModal} className='movie-thumbnail-more-info__button'> More info </button>
                <Modal 
                isOpen={modal}
                ariaHideApp={false} 
                >
                    <button onClick={closeModal}> Close </button>
                    {movieData ? <p> {movieData.Title}</p> : null}
                </Modal>
            </div>}
        </div>
    );
};

class Movies extends React.Component {

    displayMessage = () => {
        if(this.props.fetching){
            return <p className='message-container__message'> Loading.. </p>
        }else if(this.props.error){
            return <p className='message-container__message'> No movie found with that name. Try again! </p>
        }

        // Previous solution
        // {this.props.fetching && <p> Loading.. </p>}
        // {this.props.error && <p> There was an error..</p>}
    }

    render(){
        console.log(JSON.stringify(this.props.movies) === '[]')
        return(
            <React.Fragment>
                {this.props.movies && JSON.stringify(this.props.movies) !== '[]' && 
                <div className='movies-container-large'>
                    {this.props.movies ? this.props.movies.map((movie,index) => {
                        return <Movie key={index} title={movie.Title} year={movie.Year} type={movie.Type} image={movie.Poster} id={movie.imdbID}/>
                    }) : null}
                </div>}

                {this.props.fetching || this.props.error ? (
                    <div className='message-container'>
                        {this.displayMessage()}
                    </div>
                ) : null}
                
            </React.Fragment>
            
        );
    }
};

const mapStateToProps = (state) => ({  movies: state.movies, fetched: state.fetched, fetching: state.fetching, error: state.error });

export default connect(mapStateToProps)(Movies);