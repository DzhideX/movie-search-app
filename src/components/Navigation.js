import React from 'react';
import { getMovies } from '../actions/moviesActions';
import { store } from '../store/configureStore';
import { connect } from 'react-redux';

export class Navigation extends React.Component {

    state = {
        value: '',
    };

    updateInput = (e) => {
        const value = e.target.value;
            this.setState(prevState => ({
                value
            }));
    };

    handleSubmit = async () => {
        if(this.state.value){
            store.dispatch(getMovies(this.state.value));
        }
    }

    handleKeyPress = (target) => {
        if(this.state.value && target.charCode===13){
            store.dispatch(getMovies(this.state.value));
        }
    }

    render(){
        return(
            <div data-test='navigation' className='navigation'>
                <h3 data-test='navigation-logo' className='navigation-logo'> MovieSearch </h3>
                <div className='navigation-search'>
                    <input className='navigation-search__input' placeholder="Search for some movies.." onKeyPress={this.handleKeyPress} value={this.state.value} onChange={this.updateInput}/>
                    <button className='navigation-search__button' onClick={this.handleSubmit}> Search </button>
                </div>
            </div>
        );
    };
};

const mapStateToProps = (state) => ({  movies: state.movies, fetched: state.fetched });

export default connect(mapStateToProps)(Navigation);