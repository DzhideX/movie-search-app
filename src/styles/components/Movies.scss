.movies-container-large {
    width: 100%;
    justify-content: center;
    display: grid;
    padding: 0 2rem 2rem 2rem;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    @media(max-width: 1000px){
        grid-template-columns: 1fr 1fr 1fr;
    }
    @media(max-width: 770px){
        grid-template-columns: 1fr 1fr;
    }
    @media(max-width: 550px){
        grid-template-columns: 1fr;
    }
    justify-items: center;
}

.message-container{
    width: 100%;
    height: 20rem;
    display: flex;
    justify-content: center;
    align-items: center;
    @media(max-width: 600px){
        text-align: center;
        padding: 0 0.5rem 0 0.5rem;
    }
    .message-container__message{
        color:#71A593;
        font-size: 1.5rem;
        margin: 0;
    }
}

.movie-thumbnail{
    height: 300px;
    width: 220px;
    margin-top: 2rem;
    background: none;
    background-color: $light-brown;
    border-radius: 10px;
    .movie-thumbnail__no-info{
        border-radius: 10px;
        width: 100%;
        height: 100%;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        text-align: center;
    .movie-thumbnail-no-image-icon{
        height: 50px;
        margin-top: 90px;
        margin-bottom: 90px;
    }
    .movie-thumbnail__image{
        height: 235px;
        width: 170px;
        margin-bottom: 0.5rem;
    }
    }
    .movie-thumbnail-more-info{
        height: 100%;
        width: 100%;
        border-radius: 10px;
        background-color: $light-brown;
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        p{
            margin-bottom: 5px;
            color: #494949;
            font-size: 1rem;
            text-align: left;
            width: 75%;
        }
        .movie-thumbnail-more-info__description{
            color: $olive-green;
            font-size: 1.5rem;
        }
        .movie-thumbnail-more-info__button{
            width: 6.5rem;
            height: 2rem;
            margin-top: 20px;
            background-color: rgb(255, 255, 255);
            border-radius: 15px;
            border: 1px solid rgb(207, 207, 207);
            cursor: pointer;
            transition: 0.01s ease-in;
        }
        .movie-thumbnail-more-info__button:focus{
            transform: translateY(1px);
        }
    }
}

.movie-thumbnail-more-info-modal{
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: column;
    .movie-thumbnail-more-info-modal-top{
        width: 100%;
        display: flex;
        @media (max-width: 900px){
        flex-direction: column;
        }
        .movie-thumbnail-more-info-modal-image{
            height: 235px;
            width: 170px;
            margin: 1rem 0 0 1rem;
            @media (max-width: 900px){
                margin: 0 auto;
            }
        }
        .movie-thumbnail-more-info-modal-no-image-icon{
            height: 100px;
            width: 100px;
            margin: 1rem 0 0 1rem;
        }
        .movie-thumbnail-more-info-modal-top-right{
            display: flex;
            flex-direction: column;
            margin-top: 1rem;
            p{
                margin: 0 0 0.9rem 2rem;
                @media (max-width: 900px){
                    margin: 0 0 0.9rem 0.5rem;
                }
                @media (max-width: 500px){
                    font-size: 0.8rem;
                }
                span{
                    color: $olive-green;
                    margin-right: 0.7rem;
                }
            }
        }
    }
    .movie-thumbnail-more-info-modal-bottom{
        p{
            margin: 0.7rem 0 0.5rem 1rem;
            @media (max-width: 900px){
                margin: 0.7rem 0 0.5rem 0.5rem;
            }
            @media (max-width: 500px){
                font-size: 0.8rem;
            }
        }
        span{
            color: $olive-green;
            margin-right: 0.7rem;
        }
        .movie-thumbnail-more-info-modal-bottom-cards{
            display: flex;
            flex-direction: row;
            justify-content: space-around;
            margin-top: 1rem;
            @media (max-width: 500px){
                flex-direction: column;
            }
            .movie-thumbnail-more-info-modal-bottom-cards__card{
                display: flex;
                flex-direction: column;
                align-items: center;
                justify-content: center;
                padding: 1rem;
                .movie-thumbnail-more-info-modal-bottom-cards__rating-image{
                    height: 3rem;
                    width: 9.5rem;
                    @media (max-width: 900px){
                        height: 1.5rem;
                        width: 4.75rem;
                    }
                    @media (max-width: 500px){
                        height: 2rem;
                        width: 5.5rem;
                    }
                    
                }
                p{
                    margin:1rem 0 0 0;
                    font-size: 1.15rem;
                }
            }
        }
    }
}