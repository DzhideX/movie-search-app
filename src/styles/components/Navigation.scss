.navigation{
    background-color: $light-brown;
    width: 100%;
    height: 8.5vh;
    display: flex;
    justify-content: space-around;
    align-items: center;
    .navigation-logo{
        // margin: 0 0 0 3rem;
        font-size: 1.5rem;
        color:$olive-green;
    }
    .navigation-search{
        // margin: 0 3rem 0 0;
        .navigation-search__input{
            margin-right: 1.5rem;
            width: 20rem;
            border: none;
            height: 2rem;
            border-radius: 5rem;
            padding-left: 1rem;
        }
        .navigation-search__input:focus{
            border: 0.5px solid rgb(219, 219, 219);
        }
        .navigation-search__button {
            border: none;
            color:rgb(73, 73, 73);
            background: none;
            font-weight: bold;
        }
        .navigation-search__button:hover {
            cursor: pointer;
        }
    }
}