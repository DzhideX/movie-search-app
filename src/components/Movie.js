import React, { useState, useEffect } from 'react';
import noImage from '../images/340719-200.png';
import Modal from 'react-modal';
import axios from 'axios';
import MovieRating from './MovieRating';

const Movie = ({index,title,image,year,type,id}) => {

    const [movieData, setMovieData] = useState([]);
    useEffect(() => {
        axios.get(`http://www.omdbapi.com/?apikey=10d92ac&i=${id}`).then(response=> {
            setMovieData(response.data);
        });
        // eslint-disable-next-line
    }, []);
    const [hovered, setHovered] = useState(false);
    const toggleHover = () => setHovered(!hovered);
    const [modal, setModal] = useState(false);
    const openModal = () => {
        setModal(true);
    }
    const closeModal=  () => {
        setModal(false);
        setHovered(false);
    }

    return (
        <div className='movie-thumbnail'>
            {!hovered && <div className='movie-thumbnail__no-info' onMouseEnter={toggleHover}>
            {image === 'N/A' ? <img className='movie-thumbnail-no-image-icon' src={noImage} alt='not loaded'/> : <img className='movie-thumbnail__image'  src={image} alt={index}/>} 
            <p className='movie-thumbnail__title'> {title} </p>
            </div>}
            {hovered && 
            <div className='movie-thumbnail-more-info' onMouseLeave={toggleHover}> 
                <p className='movie-thumbnail-more-info__description'> Title: </p>
                <p> {title} </p>
                <p className='movie-thumbnail-more-info__description'> Year: </p>
                <p> {year} </p>
                <p className='movie-thumbnail-more-info__description'> Type: </p>
                <p> {type} </p>
                <button onClick={openModal} className='movie-thumbnail-more-info__button'> More info </button>
                <Modal 
                isOpen={modal}
                ariaHideApp={false}
                style={{
                    content: {
                        width: '67vw',
                        height:'33rem',
                        margin: 'auto'
                    }
                  }}
                onRequestClose={closeModal}
                >   
                    <div className='movie-thumbnail-more-info-modal'>
                        <div className='movie-thumbnail-more-info-modal-top'>
                            {image === 'N/A' ? <img className='movie-thumbnail-more-info-modal-no-image-icon' src={noImage} alt='not loaded'/> : <img className='movie-thumbnail-more-info-modal-image'  src={image} alt={index}/>} 
                            <div className='movie-thumbnail-more-info-modal-top-right'>
                                {movieData.Title !== 'N/A' && <p><span> Title: </span> {movieData.Title}</p>}
                                {movieData.Rated !== 'N/A' && <p><span> Rated: </span> {movieData.Rated}</p>}
                                {movieData.Released !== 'N/A' && <p><span> Released: </span> {movieData.Released}</p>}
                                {movieData.Runtime !== 'N/A' && <p><span> Run time: </span> {movieData.Runtime}</p>}
                                {movieData.Genre !== 'N/A' && <p><span> Genre: </span> {movieData.Genre}</p>}
                                {movieData.Writer !== 'N/A' && <p><span> Writer: </span>{movieData.Writer.substring(0,70) + '...'}</p>}
                                {movieData.Actors !== 'N/A' && <p><span>Actors: </span>{movieData.Actors}</p>}
                            </div>
                        </div>
                        <div className='movie-thumbnail-more-info-modal-bottom'>
                            {movieData.Plot !== 'N/A' && <p> <span> Plot: </span>  {movieData.Plot}</p>}
                            {movieData.Awards !== 'N/A' && <p> <span> Awards: </span>  {movieData.Awards}</p>}
                            <div className='movie-thumbnail-more-info-modal-bottom-cards'>
                                {movieData.Ratings.length > 0 && movieData.Ratings.map(rating => <MovieRating rating={rating}/>)}
                            </div>
                        </div>
                    </div>
                </Modal>
            </div>}
        </div>
    );
};

export default Movie;