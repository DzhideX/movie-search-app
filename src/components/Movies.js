import React from 'react';
import { connect } from 'react-redux';
import Movie from './Movie';

export class Movies extends React.Component {

    displayMessage = () => {
        if(this.props.fetching){
            return <p className='message-container__message'> Loading.. </p>
        }else if(this.props.error){
            return <p className='message-container__message'> No movie found with that name. Try again! </p>
        }
    }
    
    render(){
        return(
            <React.Fragment>
                { !(this.props.fetching || this.props.error) && 
                <div data-test='movies-container-large' className='movies-container-large'>
                    {this.props.movies ? this.props.movies.map((movie,index) => {
                        return <Movie key={index} title={movie.Title} year={movie.Year} type={movie.Type} image={movie.Poster} id={movie.imdbID}/>
                    }) : null}
                </div>}

                {this.props.fetching || this.props.error ? (
                    <div data-test='message-container' className='message-container'>
                        {this.displayMessage()}
                    </div>
                ) : null}
                
            </React.Fragment>
            
        );
    }
};

const mapStateToProps = (state) => ({  movies: state.movies, fetched: state.fetched, fetching: state.fetching, error: state.error });

export default connect(mapStateToProps)(Movies);