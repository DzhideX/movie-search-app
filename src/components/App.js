import React from 'react';
import Movies from './MovieSearch';
import Navigation from './Navigation';

function App() {
  return (
    <div data-test='app' className='app'>
        <Navigation />
        <Movies />
    </div>
  );
}

export default App;
