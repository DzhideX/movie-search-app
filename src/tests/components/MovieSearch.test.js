import { shallow } from 'enzyme';
import { Movies } from '../../components/MovieSearch';
import React from 'react';

test('app renders with no errors',()=>{
    const wrapper = shallow(<Movies />);   
    expect(wrapper.debug()).toMatchSnapshot();
});