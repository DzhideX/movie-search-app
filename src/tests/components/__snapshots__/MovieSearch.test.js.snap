// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders with no errors 1`] = `
"<Fragment>
  <div className=\\"movies-container-large\\" />
</Fragment>"
`;
